framework/src/modules/token/commands/cc_transfer.ts

```diff
@@ -49,6 +79,63 @@ export class CCTransferCommand extends BaseCommand {
 
 		try {
 			validator.validate(this.schema, params);
+
+			const [tokenChainID, _] = splitTokenID(params.tokenID);
+
+			if (
+				![this._ownChainID, params.receivingChainID, MAINCHAIN_ID].some(allowedChainID =>
```

Comment:
`MAINCHAIN_ID` should be computed using `[mainchainID]] = splitTokenID(this._tokenMethod.getMainchainTokenID())`